spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MultiRNNCell.scala

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity, TensorModule}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.serializer.{DataConverter, ModuleData, ModuleSerializable, ModuleSerializer}
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+/**
+ * Enable user stack multiple simple cells.
+ */
+class MultiRNNCell[T : ClassTag](val cells: Array[Cell[T]])(implicit ev: TensorNumeric[T])
+  extends Cell[T](
+    hiddensShape = cells.last.hiddensShape,
+    regularizers = cells.flatMap(_.regularizers)) {
+  // inputDim and hidDim must be the same with Recurrent
+  private val inputDim = Recurrent.inputDim
+  private val hidDim = Recurrent.hidDim
+
+  override var preTopology: TensorModule[T] = null
+
+  override var cell: AbstractModule[Activity, Activity, T] = buildModel()
+
+  def buildModel(): Sequential[T] = {
+    val seq = Sequential()
+    cells.foreach{ cell =>
+      if (cell.preTopology != null) {
+        cell.includePreTopology = true
+      }
+      seq.add(cell)
+    }
+    seq
+  }
+
+  override def updateOutput(input: Table): Table = {
+    val result = T()
+    result(inputDim) = input(inputDim)
+    val states = input(hidDim).asInstanceOf[Table]
+    val outputStates = T()
+
+    var i = 0
+    while (i < cells.length) {
+      result(hidDim) = states(i)
+      cells(i).forward(result).toTable
+      result(inputDim) = cells(i).output.toTable(inputDim)
+      outputStates(i) = cells(i).output.toTable(hidDim)
+      i += 1
+    }
+
+    result(hidDim) = outputStates
+    this.output = result
+    output
+  }
+
+  override def updateGradInput(input: Table, gradOutput: Table): Table = {
+    var i = cells.length
```

Comment:
better to set i to cells.length-1, so as to align with the updateOutput implemetation